modules/data-table/main/index.coffee

```diff
@@ -59,100 +265,182 @@ class DataTable extends hx.EventEmitter
       searchPlaceholder: hx.userFacingText('dataTable','search')
       selectedRowsText: hx.userFacingText('dataTable', 'selectedRows')
       sortByText: hx.userFacingText('dataTable','sortBy')
-    }, options)
 
-    selection = hx.select(selector).classed('hx-data-table', true)
-    content = selection.append('div').class('hx-data-table-content')
-    statusBar = selection.append('div').class('hx-data-table-status-bar')
-    footer = selection.append('div').class('hx-data-table-footer')
+      addFilterText: hx.userFacingText('dataTable', 'addFilter')
+      clearFiltersText: hx.userFacingText('dataTable', 'clearFilters')
+      anyColumnText: hx.userFacingText('dataTable', 'anyColumn')
 
-    resolvedOptions.pageSize = Math.min resolvedOptions.pageSize, 1000
+      advancedSearchText: hx.userFacingText('dataTable','advancedSearch')
+      advancedSearchPlaceholder: hx.userFacingText('dataTable', 'search')
+    }, options)
 
-    onInput = hx.debounce 200, => @filter(filterInput.value(), undefined, 'user')
+    resolvedOptions.pageSize = Math.min resolvedOptions.pageSize, 1000
```

Comment:
Because I've changed how the structure is built, it lets us define things in a better order.
Unfortunately, this seems to have messed up the diff a little...